File: beagle-declarative/src/main/kotlin/br/com/zup/beagle/declarative/widget/ui/NetworkImage.kt
package br.com.zup.beagle.declarative.widget.ui

import br.com.zup.beagle.declarative.widget.core.ImageContentMode
import br.com.zup.beagle.declarative.widget.core.Widget

data class NetworkImage(
    val url: String,
    val contentMode: ImageContentMode? = null /* = ImageContentMode.FIT_CENTER */
) : Widget

File: Sources/BeagleUI/Sources/Widgets/Image/Network/NetworkImage.swift
//
//  Copyright © 2019 Zup IT. All rights reserved.
//

import Foundation

public struct NetworkImage: Widget, HasAppearance {
    
    public let url: String
    public let contentMode: ImageContentMode?
    public let appearance: Appearance?
    
    // MARK: - Initialization
    
    public init(
        url: String,
        contentMode: ImageContentMode? = nil,
        appearance: Appearance? = nil
    ) {
        self.url = url
        self.contentMode = contentMode
        self.appearance = appearance
    }
    
}
